# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../doxygen/hotfix-missing_isatty.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

g++ -c -pipe -DENGLISH_ONLY -Wall -W -fno-exceptions -O2 -DNO_DEBUG -I../../src -I/opt/qt3/include -o obj/moc_expert.o moc/moc_exp
ert.cpp
<stdout>: In function 'void configYY_init_buffer(yy_buffer_state*, FILE*)':
<stdout>:2079: error: 'isatty' was not declared in this scope

--- ./src/config.l.orig	2006-11-23 15:43:20.000000000 -0300
+++ ./src/config.l	2006-11-23 15:56:12.000000000 -0300
@@ -23,6 +23,11 @@
 #include <ctype.h>
 #include <stdarg.h>
 
+#ifdef __cplusplus
+// including <unistd.h> doesn't help
+extern "C" int isatty (int __fd) __THROW;
+#endif
+
 #include <qfileinfo.h>
 #include <qdir.h>
 #include <qtextstream.h>
